File: solutions/javascript/pascals-triangle/1/pascals-triangle.js

// Computes the first n rows of the Pascal's triangle.
export function rows(n) {
  
  if (n < 1) return [];
  
  let result = [[1]];
  for (let r = 1; r < n; r++) {
    let row = [1];
    row[r] = 1;
    for (let c = 1; c < r; c++) {
      row[c] = result[r-1][c-1] + result[r-1][c];
    }
    result.push(row);
  }
  return result;
};

File: solutions/go/pascals-triangle/1/pascals_triangle.go
package pascal

func Triangle(n int) [][]int {
    triangle := make([][]int, n)
	for i:= 0; i < n; i++ {
        triangle[i] = make([]int, i+1)
        triangle[i][0] = 1
        triangle[i][i] = 1
        for j := 1; j < i; j++ {
            triangle[i][j] = triangle[i-1][j-1] + triangle[i-1][j]
        }
    }
    return triangle
}
